awsutil/generate_credentials.go

```diff
@@ -32,6 +33,12 @@ type CredentialsConfig struct {
 	// The session token if it is being used
 	SessionToken string
 
+	// The IAM endpoint to use; if not set will use the default
+	IamEndpoint string
```

Comment:
Nit: Should we capitalize to `IAMEndpoint` (same with `STSEndpoint`) in the same spirit that we have for `HTTPClient`?

Reply:
Yeah I guess I can do that. I wish I could change `HTTPClient` instead, but alas :-)